pkg/plugins/client.go

```diff
@@ -96,17 +107,29 @@ func abort(start time.Time, timeOff time.Duration) bool {
 	return timeOff+time.Since(start) > time.Duration(defaultTimeOut)*time.Second
 }
 
-func configureTCPTransport(tr *http.Transport, proto, addr string) {
-	// Why 32? See https://github.com/docker/docker/pull/8035.
-	timeout := 32 * time.Second
-	if proto == "unix" {
-		// No need for compression in local communications.
-		tr.DisableCompression = true
-		tr.Dial = func(_, _ string) (net.Conn, error) {
-			return net.DialTimeout(proto, addr, timeout)
+func readTLSConfig(config *sockets.TlsConfig) (*tls.Config, error) {
+	certPool := x509.NewCertPool()
+	file, err := ioutil.ReadFile(config.CA)
+	if err != nil {
+		return nil, err
+	}
+	certPool.AppendCertsFromPEM(file)
+
+	tlsConfig := &tls.Config{
+		MinVersion:         tls.VersionTLS10,
```

Comment:
Depends on what this is supposed to be connecting to. If it will always connect to code made by Docker, sure. If not, we will probably be better off by having 1.0 as the minimum.